include/triton/core/tritonserver.h

```diff
@@ -905,6 +905,28 @@ TRITONSERVER_InferenceRequestAppendInputData(
     const void* base, size_t byte_size, TRITONSERVER_MemoryType memory_type,
     int64_t memory_type_id);
 
+/// Assign a buffer of data to an input for execution on a particular device. 
+/// The buffer will be appended to any existing buffers for that input on that device.
+/// The 'inference_request' object takes ownership of the buffer and so the caller 
+/// should not modify or free the buffer until that ownership is released by
+/// 'inference_request' being deleted or by the input being removed
+/// from 'inference_request'.
```

Comment:
Should mention that the `TRITONSERVER_InferenceRequestAppendInputData()` is still needed as fallback, something like "If the execution happens on a unspecified device, the data appended by TRITONSERVER_InferenceRequestAppendInputData() will be used for execution"